core/mchat.php

```diff
@@ -1056,7 +1056,13 @@ public function assign_messages($rows, $page = '')
 
 			$message_age = time() - $row['message_time'];
 			$minutes_ago = $this->get_minutes_ago($message_age);
-			$datetime = $this->user->format_date($row['message_time'], $this->settings->cfg('mchat_date'), true);
+			// If relative time is selected, also display "today" / "yesterday", else display absolute time.
+			if ($this->settings->cfg('mchat_relative_time'))
+			{
+				$datetime = $this->user->format_date($row['message_time'], $this->settings->cfg('mchat_date'), false);
```

Comment:
It's a nice idea though, definitely something that could be useful. If you find a universal way to solve this I'll be happy to merge it. Thanks!